src/Service.Host/client/src/components/serialNumberTransactions/SerialNumberTransaction.js

```diff
@@ -27,32 +30,104 @@ function SerialNumberTransaction({
 }) {
     const [serialNumberTransaction, setSerialNumberTransaction] = useState({});
     const [prevSerialNumberTransaction, setPrevSerialNumberTransaction] = useState({});
+    const [newElements, setNewElements] = useState([]);
 
     const creating = () => editStatus === 'create';
     const editing = () => editStatus === 'edit';
     const viewing = () => editStatus === 'view';
 
     useEffect(() => {
-        if (item !== prevSerialNumberTransaction) {
+        if (!creating() && item !== prevSerialNumberTransaction) {
             setSerialNumberTransaction(item);
             setPrevSerialNumberTransaction(item);
         }
     });
 
-    const transCodeInvalid = () => !serialNumberTransaction.transCode;
+    const emptySernosTransCode = {
+        transCode: '',
+        sernosCount: '',
+        checkError: '',
+        correctValue: 0,
+        countIncrement: 0,
+        checkErrorMess: ''
+    };
+
+    const errorOptions = [
+        { displayText: 'None', id: 'N' },
+        { displayText: 'Error', id: 'E' },
+        { displayText: 'Warning', id: 'W' },
+        { displayText: '', id: ''}
```

Comment:
extremely pedantic but the lint gods demand a space before the final closing curly bracket